### PR TITLE
Add non-deprecated variants to configure native-memory size

### PR DESCRIPTION
**Pull request by fbarotov**
EE PR: https://github.com/hazelcast/hazelcast-enterprise/pull/5014

### CODE REVIEW

**Comment by fbarotov** [comment]
> it would be great if you could also switch get/setSize usages within our codebase to the new methods so we don't get deprecation warnings

Thanks for the comment Vassilis. Done :)

**Review by vbekiaris — APPROVED** [review]

File: hazelcast/src/main/java/com/hazelcast/client/config/ClientConfigXmlGenerator.java [hunk 1]
```diff
@@ -352,8 +352,8 @@ private static void serializers(XmlGenerator gen, SerializationConfig serializat
     private static void nativeMemory(XmlGenerator gen, NativeMemoryConfig nativeMemory) {
         gen.open("native-memory", "enabled", nativeMemory.isEnabled(),
                 "allocator-type", nativeMemory.getAllocatorType())
-                .node("size", null, "value", nativeMemory.getSize().getValue(),
-                        "unit", nativeMemory.getSize().getUnit())
+                .node("size", null, "value", nativeMemory.getCapacity().getValue(),
```
Comment (mdumandag) [inline hazelcast/src/main/java/com/hazelcast/client/config/ClientConfigXmlGenerator.java:355]: Are we going to update the XSD as well, introducing a new element for the `capacity`?

**Comment by ahmetmircik** [comment]
ee counterpart was closed, are we still keeping this PR open @petrpleshachkov?

**Comment by vbekiaris** [comment]
Closing in favour of #22255

**Pull request closed without merging**